### ISSUE TITLE
sendDisappearingPhoto doesn't seem to work.

### ISSUE THREAD

**Issue by fagenorn** [mgp25/Instagram-API#1438]
Hey,

Not sure if this is a known issue or not, but I was having some trouble with one of the endpoints related to "direct". Just a heads up before the release of 3.*.

When using the method `direct->sendDisappearingPhoto()`  it doesn't seem to be able to configure the uploaded image. Even when tried with different accounts.

Source Code:
```
<?php
require __DIR__.'/vendor/autoload.php';

/////// CONFIG ///////
$username = 'redacted';
$password = 'redacted';
$debug = true;
$truncatedDebug = false;
//////////////////////
/////// MEDIA ////////
$photoFilename = 'image.jpg';
$recipients= ["users" => ["1686677107"]];
//////////////////////


$ig = new \InstagramAPI\Instagram($debug, $truncatedDebug);
try {
    $ig->setUser($username, $password);
    $ig->login();
} catch (\Exception $e) {
    echo 'Something went wrong: '.$e->getMessage()."\n";
    exit(0);
}
try {
    $ig->direct->sendDisappearingPhoto($recipients, $photoFilename);
} catch (\Exception $e) {
    echo 'Something went wrong: '.$e->getMessage()."\n";
}
```

Log files:

> POST: https://i.instagram.com/api/v1/upload/photo/ → 42.88kB ← 200 46B

> RESPONSE: {"upload_id": "1501194889721", "status": "ok"}
-----------
> POST: https://i.instagram.com/api/v1/media/configure_to_story/ DATA: signed_body=1c3f4accda568aaaeaad2b1da628a8ccbef2a6baa426f18863b3986c80f633b7.{"recipient_users":"[[\"1686677107\"]]","device":{"manufacturer":"ZTE","model":"ZTE A2017U","android_version":"23","android_release":"6.0.1"},"thread_ids":"[\"0\"]","_csrftoken":"T9R2qRfHVAtgx71qqhPzgj7fQD9pI9As","client_shared_at":"1501194890","configure_mode":"2","source_type":"3","_uid":"3232737714","_uuid":"8bb303c9-499b-45fc-ba31-b9fc0199cc91","edits":{"crop_original_size":[512,512],"crop_zoom":1,"crop_center":[0,-0]},"extra":{"source_width":512,"source_height":512},"upload_id":"1501194889721","client_timestamp":"1501194887"}&ig_sig_key_version=4 ← 500 75B 

>RESPONSE: {"message": "Other media configure error: b'FvpYh0sebg'", "status": "fail"}

Formated:
```
{  
   "recipient_users":"[[\"1686677107\"]]",
   "device":{  
      "manufacturer":"ZTE",
      "model":"ZTE A2017U",
      "android_version":"23",
      "android_release":"6.0.1"
   },
   "thread_ids":"[\"0\"]",
   "_csrftoken":"T9R2qRfHVAtgx71qqhPzgj7fQD9pI9As",
   "client_shared_at":"1501194890",
   "configure_mode":"2",
   "source_type":"3",
   "_uid":"3232737714",
   "_uuid":"8bb303c9-499b-45fc-ba31-b9fc0199cc91",
   "edits":{  
      "crop_original_size":[  
         512,
         512
      ],
      "crop_zoom":1,
      "crop_center":[  
         0,
         -0
      ]
   },
   "extra":{  
      "source_width":512,
      "source_height":512
   },
   "upload_id":"1501194889721",
   "client_timestamp":"1501194887"
}
```

------

The image that was used was a square 512x512:
![](https://blog.codinghorror.com/content/images/uploads/2005/12/6a0120a85dcdae970b016769017677970b-pi.jpg)

**Comment by SteveJobzniak** [comment]
Hi. When I was researching allowed aspect ratios yesterday, I used `sendDisappearingPhoto` like 10 times, successfully.

I am pretty sure that the configure error you're getting happens because you're using the code wrong. $recipients is not supposed to have `['users' => 'userid']` every time. That's only to be used **1** time, to create a thread. After that you must use `['thread' => 'the active thread with that user']` instead. Otherwise you are trying to make new threads with a user you already have a thread with. And your configure params are telling it "recipient_users" that user, which of course will not be allowed.

**Comment by fagenorn** [comment]
Thanks for the reply.

Using `['thread' => ...]` instead of `['users' => ...]` did indeed work (sendText() for single user, createGroupThread() for multiple users)

I do find it odd that it's an option to create a thread using user ids. Do note, I did not have a thread with the users before attempting to send a disappearing picture.
In the app (android) it does seem possible to send users directly a disappearing picture, without having to send a message first.

Currently, I am running some tests to see why `sendDisappearingPhoto()` is failing when using user ids (single or even multiple)

**Comment by SteveJobzniak** [comment]
The reason we have user id support for all functions is because that is how the app does it... it creates the thread with a "send something to user-id" or "create 3+ people group chat" call, and after that it uses the thread ID to refer to that discussion. @valga knows more.

**Comment by valga** [comment]
Fixed in #1440: we must not use "0" as thread ids when sending story to specific users.

**Comment by SteveJobzniak** [comment]
The fix has now been merged into dev-master. So now you should be able to create new threads when sending stories to specific people, without needing to send other messages first.

Remember to thank the brilliant @valga for that :)
